core/flyout_vertical.js

```diff
@@ -332,16 +333,54 @@ Blockly.VerticalFlyout.prototype.scrollToStart = function() {
 };
 
 /**
+ * Scroll the flyout to a position.
+ * @param {Number} pos The targeted scroll position.
+ */
+Blockly.VerticalFlyout.prototype.scrollTo = function(pos) {
+  this.scrollTarget = pos * this.workspace_.scale;
+
+  // Make sure not to set the scroll target below the lowest point we can
+  // scroll to, i.e. the content height minus the view height
+  var contentHeight = this.workspace_.getMetrics().contentHeight;
+  var viewHeight = this.workspace_.getMetrics().viewHeight;
+  this.scrollTarget = Math.min(this.scrollTarget, contentHeight - viewHeight);
+
+  this.step();
+};
+
+/**
+ * Step the scrolling animation by scrolling a fraction of the way to
+ * a scroll target, and request the next frame if necessary.
+ */
+Blockly.VerticalFlyout.prototype.step = function() {
```

Comment:
Same comments as on the horizontal version.  it looks like you may just be able to combine the horizontal and vertical versions into a helper function in flyout_base.  You would pass in either `this.workspace_.scrollY` or `this.workspace_.scrollX` in the horizontal and vertical implementations.

Reply:
Yes! Did it. I used the horizontalLayout_ property in flyout_base to switch between scrollX and scrollY.